modules/apps/29-fee/ibc_module_test.go

```diff
@@ -468,3 +469,330 @@ func (suite *FeeTestSuite) TestOnChanCloseConfirm() {
+		{
+			"ack wrong format",
+			func() {},
+			channeltypes.NewResultAcknowledgement([]byte{1}).Acknowledgement(),
+			false,
+		},
+		{
+			"channel is not fee not enabled, success",
+			func() {
+				suite.chainA.GetSimApp().IBCFeeKeeper.DeleteFeeEnabled(suite.chainA.GetContext(), suite.path.EndpointA.ChannelConfig.PortID, suite.path.EndpointA.ChannelID)
+			},
+			channeltypes.NewResultAcknowledgement([]byte{1}).Acknowledgement(),
+			true,
+		},
+		{
+			"channel is not fee enabled, fail",
+			func() {
+				suite.chainA.GetSimApp().IBCFeeKeeper.DeleteFeeEnabled(suite.chainA.GetContext(), suite.path.EndpointA.ChannelConfig.PortID, suite.path.EndpointA.ChannelID)
+			},
+			types.IncentivizedAcknowledgement{
+				Result:                channeltypes.NewResultAcknowledgement([]byte{1}).Acknowledgement(),
+				ForwardRelayerAddress: suite.chainB.SenderAccount.GetAddress().String(),
+			}.Acknowledgement(),
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		suite.Run(tc.name, func() {
+			suite.SetupTest()
+
+			// open incentivized channel
+			suite.coordinator.Setup(suite.path)
+
+			// set up coins
+			coin := ibctesting.TestCoin
+			fungibleTokenPacket := transfertypes.NewFungibleTokenPacketData(
+				coin.Denom,
+				sdk.NewInt(100).Uint64(),
+				suite.chainA.SenderAccount.GetAddress().String(),
+				suite.chainB.SenderAccount.GetAddress().String(),
+			)
+			packet := channeltypes.NewPacket(
+				fungibleTokenPacket.GetBytes(),
+				suite.chainA.SenderAccount.GetSequence(),
+				suite.path.EndpointA.ChannelConfig.PortID,
+				suite.path.EndpointA.ChannelID,
+				suite.path.EndpointB.ChannelConfig.PortID,
+				suite.path.EndpointB.ChannelID,
+				clienttypes.NewHeight(0, 100),
+				0,
+			)
+
+			// set up module and callbacks
+			module, _, err := suite.chainA.App.GetIBCKeeper().PortKeeper.LookupModuleByPort(suite.chainA.GetContext(), ibctesting.TransferPort)
+			suite.Require().NoError(err)
+
+			cbs, ok := suite.chainA.App.GetIBCKeeper().Router.GetRoute(module)
+			suite.Require().True(ok)
+
+			packetId := channeltypes.NewPacketId(suite.path.EndpointA.ChannelID, suite.path.EndpointA.ChannelConfig.PortID, suite.chainA.SenderAccount.GetSequence())
+			identifiedFee := types.NewIdentifiedPacketFee(
+				packetId,
+				types.Fee{
+					ReceiveFee: validCoins,
+					AckFee:     validCoins2,
+					TimeoutFee: validCoins3,
+				},
+				suite.chainA.SenderAccount.GetAddress().String(),
+				[]string{},
+			)
+			err = suite.chainA.GetSimApp().IBCFeeKeeper.EscrowPacketFee(suite.chainA.GetContext(), identifiedFee)
+			suite.Require().NoError(err)
+
+			// malleate test case
+			tc.malleate()
+
+			err = cbs.OnAcknowledgementPacket(suite.chainA.GetContext(), packet, tc.ackBytes, suite.chainB.SenderAccount.GetAddress())
+
+			if tc.expPass {
+				suite.Require().NoError(err, "unexpected error for case: %s", tc.name)
+			}
+
+			if !tc.expPass {
+				suite.Require().Error(err, "%s expected error but returned none", tc.name)
+			}
```

Comment:
```suggestion
			if tc.expPass {
				suite.Require().NoError(err, "unexpected error for case: %s", tc.name)
			} else {
			        suite.Require().Error(err, "%s expected error but returned none", tc.name)
			}
```
consistency improves readability (this is how all our tests are structured). Optionally the error message can be removed:
```
suite.Require().Error(err)
```

The `suite.Run` will print the test case name